doc: set up e2e test environment for IAS token (#1552)
SAP/cloud-sdk-js: knowledge-base/how-tos/0006-set-up-identity-service.md

> ## Set up identity service on the CF

Reviewer: ```suggestion
## Set up identity service (IAS) on CF
```

As SDK developers, we would like to have an e2e test running on the CF for testing the IAS features.

> This documentation shows how to set up such e2e test landscape.

Reviewer: ```suggestion
This documentation shows how to set up an e2e test landscape.
```

### You will learn

> After reading this tutorial, you will be able to:

Reviewer: maybe "how-to" instead of tutorial?

- create an identity service
- create a service key for the identity service
- manage the default identity provider
- verify the IAS token (JWT) from the identity service

### Prerequisites

> This documentation assumes you know how to set up a minimal e2e test landscape on the CF, which includes:

Reviewer: ```suggestion
This documentation assumes you know how to set up a minimal e2e test landscape on CF, which includes:
```

> - a space
> - xsuaa service
> - saas registry service
> - app router set up and deployment
> - your application set up and deployment

Reviewer: Maybe make the list more parallel
```suggestion
- creating a space
- creating an xsuaa service instance
- creating a saas registry service instance
- configuring and deploying an app router
- configuring and deploying an application
```

> ### Step 1: Create identity service on the CF

Reviewer: ```suggestion
### Step 1: Create an identity service instance on CF
```

{
  "oauth2-configuration": {
    "redirect-uris": ["https://*.cfapps.sap.hana.ondemand.com/login/callback"]
  },
  "consumed-services": [],
  "xsuaa-cross-consumption": true,
  "multi-tenant": true
}
```

### Step 2: Create a service key for the identity service

Similarly, you can use the command below:

```
cf create-service-key your-identity-service-name your-identity-service-key-name
```

You can check your service key information via the command below:

```
cf service-key your-identity-service-name your-identity-service-key-name
```

The response should look like the json below, where the `url` shows how to access the default IdP:

```json
{
  "clientid": "xxx",
  "clientsecret": "xxx",
  "domain": "xxx.ondemand.com",
  "domains": ["xxx.ondemand.com"],
  "url": "https://xxx.xxx.ondemand.com",
  "zone_uuid": "xxx"
}
```

### Step 3: Manage the default IdP

Follow the steps below for the IdP management:

1. Go to the admin page (`/admin`) of the default IdP shown in the previous step like: `http://xx.xx.ondemand.com/admin/`.
1. Log in with your admin account.
1. Click `Applications & Resources`.
1. Click `Applications`.
1. Search for your identity service name and select it.

> 1. Make sure the `Multitenant Application` is checked.

Reviewer: ```suggestion
1. Make sure `Multitenant Application` is checked.
```

1. You can configure other settings of the default IdP.

### Step 4 Verify the IAS token from the identity service

I assume you have a real application (e.g., VDM E2E test application) with an app router.

#### Application deployment

Add the identity service created to the `manifest.yaml` for service binding.
Please note, as the identity service can be used without an XSUAA service, we don't have to bind the XSUAA service for now.

#### App Router deployment

##### Service binding

Similarly, add the same identity service to the `manifest.yaml`.

##### Configure `route` and `TENANT_HOST_PATTERN`

> The `route` url looks like below, where the `tenant id` part of the URL has to be the `tenant id` on the CF (with the GUID pattern).

Reviewer: ```suggestion
The `route` URL looks like below, where the `tenant id` part of the URL has to be the `tenant id` on the CF (with the GUID pattern).
```
